addons/test_website/controllers/main.py

```diff
@@ -119,3 +119,7 @@ def get_post_method(self, **kw):
     @http.route(['/get_post_nomultilang'], type='http', auth="public", methods=['GET', 'POST'], website=True, multilang=False, sitemap=False)
     def get_post_method_no_multilang(self, **kw):
         return request.make_response('get_post_nomultilang')
+
+    @http.route(['/test_route_1'], type='*', auth="public", methods=['GET', 'POST'], csrf=False)
+    def test_route_1(self, **kw):
+        return str(kw) + str(request.jsonrequest if hasattr(request, 'jsonrequest') else '')
```

Comment:
Can't you do a if/else based on the hasattr instead?
Would be easier to understand that you return one parameter or the other based on the type.

Also, is there no cleaner way to check the request type?

(I assume that "kw" is empty when you access the route with a json type).